warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexFunctionVisitor.java

```diff
@@ -66,34 +66,34 @@ public Object visit(ASTFunctionNode node, Object data) {
                 JexlNode regexNode = buildRegexNode(identifier, functionMetadata.name(), arguments.get(1).image);
                 if (regexNode != null) {
                     children.add(regexNode);
+                } else {
+                    // failure to rewrite a regex means a copy of the original node is returned
+                    return returnNode;
```

Comment:
this also is a change in behavior? otherwise is covered in the case 0 below

Reply:
You can absolutely have multiple fields in a function node - this is very common with geo function queries.  You are also skipping all of the trace statements in the subsequent switch statement by allowing the code to return at this point.  I think you should remove the else statement.  

Also, as a general comment, with this being advertised as a sonar lint PR, I think it's kind of bad practice to mix in functionality changes as well.

Reply:
Strictly speaking this wasn't a functional change -- the original node would still be returned if the number of rebuilt regex nodes did not match the number of identifiers. I did misspeak (mistype?) when I said there would only be one identifier -- there is often more than one identifier.

Regardless, I'll revert this change so we keep the trace logging as part of the execution.